lib/build/recipe/session/cookieAndHeaders.js

```diff
@@ -1,69 +1,53 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+const constants_1 = require("../../constants");
+const authorizationHeaderKey = "authorization";
 const accessTokenCookieKey = "sAccessToken";
+const accessTokenheaderKey = "st-access-token";
 const refreshTokenCookieKey = "sRefreshToken";
+const refreshTokenHeaderKey = "st-refresh-token";
 // there are two of them because one is used by the server to check if the user is logged in and the other is checked by the frontend to see if the user is logged in.
 const idRefreshTokenCookieKey = "sIdRefreshToken";
-const idRefreshTokenHeaderKey = "id-refresh-token";
+const idRefreshTokenHeaderKey = "st-id-refresh-token";
 const antiCsrfHeaderKey = "anti-csrf";
-const ridHeaderKey = "rid";
 const frontTokenHeaderKey = "front-token";
 /**
  * @description clears all the auth cookies from the response
  */
-function clearSessionFromCookie(config, res) {
-    setCookie(config, res, accessTokenCookieKey, "", 0, "accessTokenPath");
-    setCookie(config, res, refreshTokenCookieKey, "", 0, "refreshTokenPath");
-    setCookie(config, res, idRefreshTokenCookieKey, "", 0, "accessTokenPath");
-    res.setHeader(idRefreshTokenHeaderKey, "remove", false);
-    res.setHeader("Access-Control-Expose-Headers", idRefreshTokenHeaderKey, true);
-}
-exports.clearSessionFromCookie = clearSessionFromCookie;
-/**
- * @param expiry: must be time in milliseconds from epoch time.
- */
-function attachAccessTokenToCookie(config, res, token, expiry) {
-    setCookie(config, res, accessTokenCookieKey, token, expiry, "accessTokenPath");
-}
-exports.attachAccessTokenToCookie = attachAccessTokenToCookie;
-/**
- * @param expiry: must be time in milliseconds from epoch time.
- */
-function attachRefreshTokenToCookie(config, res, token, expiry) {
-    setCookie(config, res, refreshTokenCookieKey, token, expiry, "refreshTokenPath");
-}
-exports.attachRefreshTokenToCookie = attachRefreshTokenToCookie;
-function getAccessTokenFromCookie(req) {
-    return req.getCookieValue(accessTokenCookieKey);
-}
-exports.getAccessTokenFromCookie = getAccessTokenFromCookie;
-function getRefreshTokenFromCookie(req) {
-    return req.getCookieValue(refreshTokenCookieKey);
+function clearSession(config, req, res, userContext) {
+    const transferMethod = config.getTokenTransferMethod({ req, userContext });
+    const tokenTypes = ["access", "refresh", "idRefresh"];
+    for (const token of tokenTypes) {
+        setToken(config, req, res, token, "", 0, userContext, transferMethod);
+        if (transferMethod === "header" && hasTokenCookie(req, token)) {
+            setToken(config, req, res, token, "", 0, userContext, "cookie");
+        }
```

Comment:
add comment explaining that this if statement is only for migration from cookie based on header based.